### PR TITLE
feat: Record "around" method

### PR DESCRIPTION
**Pull request by kgilpin**
Methods labeled `job.perform`, `cli.command`, `message.handle` will automatically start and end a recording, if `APPMAP_RECORD_REQUESTS` is enabled.

Fixes https://github.com/getappmap/appmap-ruby/issues/345

### CODE REVIEW

**Review by apotterri — CHANGES_REQUESTED** [review]
Code looks good. Please clean up the history before merging:

* typo in the subject for commit [2c370fe](https://github.com/getappmap/appmap-ruby/pull/348/commits/2c370fe788d314ecf827fd8b0964359e9a6c3a9b)
* Re [f6bca90](https://github.com/getappmap/appmap-ruby/pull/348/commits/f6bca90ee2a6629c0cf5ba53da1ceadfb5002131): I'd rather see tests for a feature in the same commit as the feature.
* `fixup!` commit here: [042459a](https://github.com/getappmap/appmap-ruby/pull/348/commits/042459a45878cbdcdbe41ddfaba3a4e4032e055f)
* There are two separate formatting commits. Maybe they can be combined? If not, they should both have the same label (I'd vote for `refactor` over `chore`).

File: spec/around_recording_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      expect(appmap_file).to match(/perform_now_\d+_\d+\.appmap\.json\z/)
+      appmap = JSON.parse(File.read(appmap_file))
+      metadata = appmap["metadata"]
+      expect(metadata["recorder"]).to eq(
+        "name" => "command",
+        "type" => "requests"
+      )
+    end
+  end
+end
```
Comment (dividedmind) [inline spec/around_recording_spec.rb:32]: The commit with the test (f6bca90ee2a6629c0) should probably be merged with the previous (`feat`) one.

**Review by dividedmind — APPROVED** [review]
I wonder if we should consider dropping support for Ruby 2.

File: .standard.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+ruby_version: 2.6
```
[the remark this answers is not in the document]
Comment (kgilpin) [reply .standard.yml:1]: I mean, it still works so I'm inclined to keep it.

File: lib/appmap/hook/record_around.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+          tracer = @tracer
+          @tracer = nil
+          AppMap.tracing.delete(tracer)
+
+          events = tracer.events.map(&:to_h)
+
+          timestamp = DateTime.now
+          appmap_name = "#{hook_method.name} (#{Thread.current.object_id}) - #{timestamp.strftime("%T.%L")}"
+          appmap_file_name = AppMap::Util.scenario_filename([hook_method.name, Thread.current.object_id, Time.now.nsec].join("_"))
```
[the remark this answers is not in the document]
Comment (kgilpin) [reply lib/appmap/hook/record_around.rb]: Maybe but for now I am going for consistency with requests recording.

File: lib/appmap/metadata.rb [hunk 4]
```diff
@@ -9,20 +9,21 @@ def detect
...
             version: RUBY_VERSION
           },
           client: {
-            name: 'appmap',
+            name: "appmap",
             url: AppMap::URL,
             version: AppMap::VERSION
-          }
+          },
+          timestamp: Time.now.to_f
```
Comment (dividedmind) [inline lib/appmap/metadata.rb:21]: ISO timestamp would be much more useful.

Comment (kgilpin) [reply lib/appmap/metadata.rb:21]: Why? It’s the same information.

Comment (dividedmind) [reply lib/appmap/metadata.rb]: Because numeric timestamps are utterly meaningless to a human :)

**Pull request merged**